pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -45,6 +39,7 @@ def __init__(self) -> None:
         # the current split index when the batch gets split into chunks in truncated backprop through time
         self.split_idx: Optional[int] = None
         self.optimizer_loop = OptimizerLoop()
+        self.manual_loop = ManualOptimization()
```

Comment:
this is a bit confusing, this looks like we have two loops for the optimizer? how they interact?

Reply:
they don't interact. automatic optimization has an optimizer loop and manual optimization does not. so either one or the other is active, never both

Reply:
so would it be cleaner to have also assigned only one of them?

Reply:
@Borda Do you mean something like this: 

```py
optimization_loop = OptimizerLoop() if automatic_optimization else ManualOptimization()
```

Reply:
yes :rabbit:

Reply:
this information is not available at the time of instantiation. manual_optimization True/False is defined on the LightningModule, therefore this will only be known at the time when fit is called.

Same logic applies also to TBPTT where we have the batch loop even when we don't actually know that `model.truncated_bptt_steps > 0`.

Reply:
IMO, I prefer the logic introduced by @awaelchli, as the code is well separated.
It would be possible in the future to get more lower-level loop.